### ISSUE TITLE
False positive in UnnecessaryAbstractClass

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4093]
```kotlin
public abstract class Comonent {
    internal abstract fun getInjector(): AnInternalClass
}

internal class AnInternalClass()
```

`Component` can't be replaced by an interface because I want `Component` to be public and `getter` to be internal and kotlin doesn't allow to have a public interface with internal functions.

## Context
My snippet seems kind of silly but that pattern happens when you uses Dagger. You want to expose your `Component` because it should be created from outside your module (you don't care about creating an instance of it). But the injectors that it provides are internal because it expose internal classe that no one from outside needs to know.

## Your Environment
* Version of detekt used: 1.18.1

**Comment by cortinico** [comment]
> because I want `Bar` to be public and `getter` to be internal

Seems like `Bar` is missing in your example

**Comment by BraisGabin** [comment]
My bad! I just editted it. I was using "Bar" and "Foo" and then realised that proper names would help to understand the use case and changed it. But I didn't change the references. This with IntelliJ doesn't happen 😂
